### ISSUE TITLE
ImageDataGenerator.mean returns a single float value with featurewise_center=True

### ISSUE THREAD

**Issue by ahmedhosny** [keras-team/keras#5581]
From my understanding, the featurewise_center flag subtracts the **mean image** from every image in the dataset. However, ImageDataGenerator.mean and ImageDataGenerator.std returns a single float value. 
For say 9 grayscale images of 120x120 dimension, I expect the mean image which will be "featurewise" subtracted from each image to be of size [120,120,1] 
```
#
datagenTrain = ImageDataGenerator(
featurewise_center=True, 
featurewise_std_normalization=True,
samplewise_center=False,
samplewise_std_normalization=False)  
datagenTrain.fit(myData) 
#
print myData.shape
print datagenTrain.mean
print datagenTrain.std

(9, 120, 120, 1)
[[[-145.6569519]]]
[[[ 356.71228027]]]
```

**Comment by TropComplique** [comment]
I have a similar problem.
```
data_generator = ImageDataGenerator(
    featurewise_center=True
)
data_generator.fit(X_train)
data_generator.mean.shape
>>> (1, 1, 3)
X_train.shape
>>> (20607, 32, 32, 3)
```
But the mean must be of shape `(32, 32, 3)`.

**Comment by TropComplique** [comment]
Well, a solution is to do it by hand:
```
X_train = X_train - X_train.mean(0)
X_train = X_train/X_train.std(0)
```

**Comment by TropComplique** [comment]
Hi @ahmedhosny. Please see the quote from this question: https://stackoverflow.com/questions/38553927/batch-normalization-in-convolutional-neural-network
>For convolutional layers, we additionally want the normalization to obey the convolutional property – so that different elements of the same feature map, at different locations, are normalized in the same way.

I believe that for images we're normalizing all dimensions but last one (channel).

**Comment by alkamid** [comment]
I think the naming here is a bit unfortunate. Looking at the code in https://github.com/fchollet/keras/blob/master/keras/preprocessing/image.py, it's definitely per-channel:

```python
if self.featurewise_center:
   self.mean = np.mean(x, axis=(0, self.row_axis, self.col_axis))
   broadcast_shape = [1, 1, 1]
   broadcast_shape[self.channel_axis - 1] = x.shape[self.channel_axis]
   self.mean = np.reshape(self.mean, broadcast_shape)
   x -= self.mean
```

I would call it **channel-wise centering** rather than feature-wise, which IMO should calculate the mean of every pixel across the dataset.

**Comment by MiguelM77** [comment]
Those operations (featurewise) means global mean/std/zca operation on yours whole data set
